data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -1026,7 +1027,8 @@ public int getParameterIndex() {
 
             for (PersistentProperty prop : persistentProperties) {
                 traversePersistentProperties(prop, (associations, property) -> {
-                    if (prop.isGenerated()) {
+                    boolean generated = PersistentEntityUtils.isPropertyGenerated(entity, prop, property);
```

Comment:
Can we somehow fix the prop.isGenerated to return true?

Reply:
Does not look like. We have for example `Author(id)` which is generated in context of `Author` entity insert/update. But, when doing `Book` insert/update then it is not generated and it needs to be updatable. This is why we need more context to check if field is generated ie. involved in insert/update.